ts/Grid/Pro/CreditsComposition.ts

```diff
@@ -0,0 +1,174 @@
+/* *
+ *
+ *  Grid Credits class
+ *
+ *  (c) 2020-2024 Highsoft AS
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ *  Authors:
+ *  - Dawid Dragula
+ *  - Sebastian Bochan
+ *
+ * */
+
+'use strict';
+
+/* *
+ *
+ *  Imports
+ *
+ * */
+
+import type { CreditsOptions } from '../Core/Options';
+import type Grid from '../Core/Grid';
+
+import Credits from '../Core/Credits.js';
+import U from '../../Core/Utilities.js';
+
+const {
+    addEvent,
+    merge
+} = U;
+
+/* *
+ *
+ *  Class
+ *
+ * */
+
+/**
+ * Represents a credits in the data grid.
+ */
+class CreditsComposition extends Credits {
+    /* *
+    *
+    *  Methods
+    *
+    * */
+
+    /**
+     * Set the content of the credits.
+     */
+    private setContent(): void {
+        const { text, href } = this.options;
+
+        this.textElement.innerHTML = text || '';
+        this.textElement.setAttribute('href', href || '');
+    }
+
+    /**
+     * Append the credits to the container. The position of the credits is
+     * determined by the `position` option.
+     */
+    private appendToContainer(): void {
+        const { position } = this.options;
+
+        if (position === 'top') {
+            // Append the credits to the top of the table.
+            this.grid.contentWrapper?.prepend(this.containerElement);
+            return;
+        }
+
+        // Append the credits to the bottom of the table.
+        this.grid.contentWrapper?.appendChild(this.containerElement);
```

Comment:
I know it's a bit out of scope and that it was already like that in the original, but since we can simplify it, I suggest to make it shorter:

```ts
this.grid.contentWrapper?.[position === 'top' ? 'prepend' : 'appendChild'](this.containerElement);
```

Reply:
Is it much more transparent? Well...